File: frontend/src/api/poet.ts
import request from '@/utils/request'

export interface PoetBrief {
  id: number
  name: string
  dynasty: string
  alias?: string
  birth_death_desc?: string
  styles?: string
  brief?: string
  representative_works?: string
  influence_score: number
}

export interface PoetListItem {
  id: number
  name: string
  dynasty: string
  alias?: string
  birth_death_desc?: string
  styles?: string
  brief?: string
  influence_score: number
  poem_count: number
}

export interface PoetListResponse {
  items: PoetListItem[]
  total: number
}

export interface PoetDetail extends PoetBrief {
  birth_year?: string
  death_year?: string
  detailed_bio?: string
  poem_count: number
  portrait_url?: string
  created_at?: string
  updated_at?: string
}

export const poetApi = {
  getPoets(params: { page?: number; page_size?: number; dynasty?: string; keyword?: string } = {}) {
    return request.get<PoetListResponse>('/poets', { params })
  },

  getByName(name: string, dynasty?: string) {
    const params: Record<string, string> = { name }
    if (dynasty) params.dynasty = dynasty
    return request.get<PoetBrief | null>('/poets/by-name', { params })
  },

  getDetail(id: number) {
    return request.get<PoetDetail>(`/poets/${id}`)
  },
}

File: backend/app/schemas/poet.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class PoetBase(BaseModel):
    name: str
    dynasty: str
    alias: Optional[str] = None
    birth_year: Optional[str] = None
    death_year: Optional[str] = None
    birth_death_desc: Optional[str] = None
    styles: Optional[str] = None
    brief: Optional[str] = None
    detailed_bio: Optional[str] = None
    representative_works: Optional[str] = None
    influence_score: int = 50
    portrait_url: Optional[str] = None


class PoetCreate(PoetBase):
    pass


class PoetDetail(PoetBase):
    id: int
    poem_count: int = 0
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class PoetListItem(BaseModel):
    id: int
    name: str
    dynasty: str
    alias: Optional[str] = None
    birth_death_desc: Optional[str] = None
    styles: Optional[str] = None
    brief: Optional[str] = None
    influence_score: int = 50
    poem_count: int = 0

    class Config:
        from_attributes = True


class PoetListResponse(BaseModel):
    items: List[PoetListItem]
    total: int


class PoetBriefForPoem(BaseModel):
    id: int
    name: str
    dynasty: str
    alias: Optional[str] = None
    birth_death_desc: Optional[str] = None
    styles: Optional[str] = None
    brief: Optional[str] = None
    representative_works: Optional[str] = None
    influence_score: int = 50

    class Config:
        from_attributes = True
